protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -118,38 +162,115 @@ open class CodeGenerator(
         // (e.g. `pbandk.wkt.Value.Companion`) is automatically frozen because it's a singleton. But Kotlin/Native
+
+        val allFields = type.sortedStandardFieldsWithOneOfs()
+        val chunkSize = 200
+        val needToChunk = allFields.size > chunkSize
+
         // doesn't allow cyclic frozen structures:
         // https://kotlinlang.org/docs/reference/native/concurrency.html#global-variables-and-singletons. In order to
         // break the circular references, `descriptor` needs to be a `lazy` field.
         line("override val descriptor: pbandk.MessageDescriptor<$fullTypeName> by lazy {").indented {
             // XXX: When a message has lots of fields (e.g. `TestAllTypesProto3`), declaring the list of field
             // descriptors directly in the [MessageDescriptor] constructor can cause a
             // `java.lang.OutOfMemoryError: Java heap space` error in the Kotlin compiler (as of Kotlin 1.4.10).
-            // As a workaround, we declare the list of fields as a separate variable rather than inline in the
-            // constructor. We also use a list builder rather than `listOf()` in order to have lots of small statements
-            // instead of a giant expression (which the compiler struggles with).
-            val allFields = type.sortedStandardFieldsWithOneOfs()
-            line("val fieldsList = ArrayList<pbandk.FieldDescriptor<$fullTypeName, *>>(${allFields.size}).apply {").indented {
-                allFields.forEach { (field, oneof) ->
-                    line("add(").indented {
-                        line("pbandk.FieldDescriptor(").indented {
-                            line("messageDescriptor = this@Companion::descriptor,")
-                            line("name = \"${field.name}\",")
-                            line("number = ${field.number},")
-                            line("type = ${field.fieldDescriptorType(oneof != null)},")
-                            oneof?.let { line("oneofMember = true,") }
-                            field.jsonName?.let { line("jsonName = \"$it\",") }
-                            line("value = $fullTypeName::${field.kotlinFieldName}")
-                        }.line(")")
-                    }.line(")")
+            // As a workaround, we generate methods to generate each fieldDescriptor in chunks, as many as needed, with
+            // a max size of $chunkSize to limit the size of the methods.
+            if (needToChunk) {
+                line("val fieldsList = ArrayList<pbandk.FieldDescriptor<$fullTypeName, *>>(${allFields.size})")
+                allFields.chunked(chunkSize).forEachIndexed { index, _ ->
+                    line("addFields${index}(fieldsList)")
                 }
-            }.line("}")
+            } else {
+                addFields(
+                    allFields,
+                    fullTypeName,
+                    line("val fieldsList = ArrayList<pbandk.FieldDescriptor<$fullTypeName, *>>(${allFields.size}).apply {")
```

Comment:
Passing the `CodeGenerator` to the `addFields()` method is weird since that method is already inside the `CodeGenerator` class. It'd be clearer if you just passed in the `String` value and called `line()` inside of `addFields()`.

Though I think this could be simplified even more by declaring `fieldsList` outside of this if/else block. I.e.:

```kt
    ...
    line("val fieldsList = ArrayList<pbandk.FieldDescriptor<$fullTypeName, *>>(${allFields.size})")
    if (needToChunk) {
        allFields.chunks(chunkSize).forEachIndexed { index, _ ->
            line("addFields${index}(fieldsList)")
        }
    } else {
        addFields(allFields, fullTypeName)
    }
    ...
    if (needToChunk) {
        allFields.chunked(chunkSize).forEachIndexed { index, chunk ->
            line("fun addFields${index}(fieldsList: ArrayList<pbandk.FieldDescriptor<$fullTypeName, *>>) {").indented {
                addFields(chunk, fullTypeName)
            }.line("}")
        }
    }
}

private fun addFields(
    chunk: List<Pair<File.Field.Numbered, File.Field.OneOf?>>,
    fullTypeName: String
) {
    line("fieldsList.apply {").indented {
        chunk.forEach { (field, oneof) ->
            line("add(").indented {
                line("pbandk.FieldDescriptor(").indented {
                    generateFieldDescriptorConstructorValues(
                        field,
                        fullTypeName,
                        oneof,
                        "this@Companion::descriptor"
                    )
                }.line(")")
            }.line(")")
        }
    }.line("}")
}
```

Reply:
Love the suggestion, my initial attempt was to preserve changing a bunch of generate files with the placement of `apply` but I certainly like your solution better!